React/Views/RCTRefreshControl.m

```diff
@@ -11,18 +11,55 @@
 
 #import "RCTUtils.h"
 
-@implementation RCTRefreshControl
+@implementation RCTRefreshControl {
+  BOOL _initialRefreshingState;
+  BOOL _isInitialRender;
+}
 
 - (instancetype)init
 {
   if ((self = [super init])) {
     [self addTarget:self action:@selector(refreshControlValueChanged) forControlEvents:UIControlEventValueChanged];
+    _isInitialRender = true;
   }
   return self;
 }
 
 RCT_NOT_IMPLEMENTED(- (instancetype)initWithCoder:(NSCoder *)aDecoder)
 
+- (void)layoutSubviews
+{
+  [super layoutSubviews];
+  
+  // If the control is refreshing when mounted we need to call
+  // beginRefreshing in layoutSubview or it doesn't work.
+  if (_isInitialRender && _initialRefreshingState) {
+    [self beginRefreshing];
+  }
+  _isInitialRender = false;
+}
+
+- (void)beginRefreshing
+{
+  // When using begin refreshing we need to adjust the ScrollView content offset manually.
+  UIScrollView *scrollView = (UIScrollView *)self.superview;
+  CGPoint offset = CGPointMake(scrollView.contentOffset.x, scrollView.contentOffset.y - self.frame.size.height);
+  // Don't animate when the prop is set initialy.
+  if (_isInitialRender) {
+    [scrollView setContentOffset:offset animated:NO];
+    [super beginRefreshing];
+  } else {
+    [UIView animateWithDuration:0.25
```

Comment:
The reason is that [super beginRefreshing] must be called after the animation is over otherwise it didn't work properly. I'll add a comment to explain this.